src/libsystemd/sd-device/device-private.c

```diff
@@ -468,6 +468,12 @@ int device_new_from_nulstr(sd_device **ret, uint8_t *nulstr, size_t len) {
 
                 i += end - key + 1;
 
+                /* netlink messages for some devices contain an unwanted newline at the end of value.
+                 * Let's drop the newline and remaining strings after the newline if exists. */
+                newline = strchr(key, '\n');
+                if (newline)
+                        *newline = '\0';
```

Comment:
hmm, this truncates at the first newline. Shouldn't we more rather drop \n only from the actual end of the string?

```c
delete_trailing_chars(key, "\n");
``

Reply:
I have found no case that a nulstr contains a new line in a middle of the string. However, if a nulstr has trailing newline like `MODALIAS=xxxxxx\n\0`, then the corresponding uevent file contains an empty line:
```
MODALIAS=xxxxxx

```
So, I have not confirmed, but it seems that kernel maps nulstr to uevent file. Then, if a nulstr contains a newline in a middle of the string like `MODALIAS=xxx\nyyy\0`, then the corresponding uevent file may become the following:
```
MODALIAS=xxx
yyy
```
In such a case, `device_read_uevent_file()` ignore `yyy` and set `MODALIAS=` to `xxx`. So, `device_new_from_nulstr()` should do the same. That is, drop any characters after new line.

Reply:
Yeah, I think it's reasonable to drop anything after the newline.